From patchwork Wed May  4 21:15:31 2016
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [U-Boot,3/6] arm64: sunxi: reserve space for boot0 header
From: Andre Przywara <andre.przywara@arm.com>
X-Patchwork-Id: 618678
Message-Id: <1462396534-32390-4-git-send-email-andre.przywara@arm.com>
To: Hans de Goede <hdegoede@redhat.com>,
	Ian Campbell <ijc@hellion.org.uk>
Cc: Andre Przywara <andre.przywara@arm.com>, u-boot@lists.denx.de
Date: Wed,  4 May 2016 22:15:31 +0100

The Allwinner provided boot0 boot loader requires a header before the
U-Boot binary to both check its validity and to find other blobs to
load. There is a tool called boot0img which fills the header
appropriately.
Reserve some space at the beginning of the binary to later hold the
header if needed.
Please note that the header is jumped over already by U-Boot anyway,
so filling the header is optional and can be skipped if for instance
boot0 is not used.

Signed-off-by: Andre Przywara <andre.przywara@arm.com>
---
 arch/arm/cpu/armv8/start.S | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/cpu/armv8/start.S b/arch/arm/cpu/armv8/start.S
index deb44a8..b4c4867 100644
--- a/arch/arm/cpu/armv8/start.S
+++ b/arch/arm/cpu/armv8/start.S
@@ -21,6 +21,9 @@
 _start:
 	b	reset
 
+#ifdef CONFIG_ARCH_SUNXI
+	.space 0x5fc	/* can be filled with a boot0 header if needed */
+#endif
 	.align 3
 
 .globl	_TEXT_BASE
